lib/agenda/cancel.ts

```diff
@@ -1,18 +1,19 @@
-'use strict';
-const debug = require('debug')('agenda:cancel');
+import { Agenda } from './index';
```

Comment:
as you only use the type from Agenda here, you can use
"import type { Agenda } from './index'"
this helps for detecting circular dependencies.. index is importing cancel too. By just importing the type, circular dependency checks are not a problem.

Reply:
Circular dependencies aren’t an issue when typescript compiles. 🙄

Reply:
I don't understand your down vote.
Yeah it is not a general problem, it is just a more defined approach how you import and use "Agenda".
Please don't get offended, I just was offering my help!

Reply:
import type is the way to go. When transpiled to javascript the import will be dropped as it is marked as type only import.

Reply:
If setup correctly typescript it can detect that without the need for the explicit “type import”. We shouldn’t be replying on a dev to add “type” before an import to solve this instead we should setup our config correctly.

Reply:
That's not true, no idea where you found this information. It has nothing todo with the setup itself, if you include a class file, there are just many cases where a compiler can not figure out if it is a type only import or not. By specifying import type you not just give the compiler the right hint,but also forbid non type usage in the file. It's super useful, look it up: https://github.com/microsoft/TypeScript/pull/35200

Reply:
Yes using tsc itself does that. We’re not going to be using tsc.

Reply:
Can you shed some light on this statement? How you gonna use typescript without typescript compiler? ;) And besides that, no offence, but it would be great to let things open to discuss, instead of deciding on the best way ahead. At least that's what is my experience with open source projects and collaboration ;)

Reply:
`tsc` is **A** compiler.

We'll be using https://github.com/egoist/tsup which will give us a es5 commonjs bundle with a definition file. The commonjs won't include the import as it's not used for anything but the type.

Reply:
typescript and esbuild is not recommended for produciton (yet). They haven't reached 1.x. https://github.com/evanw/esbuild#disclaimers

But open to new tools, especially because esbuild is super fast :) But what would be a valid reason to use a so fresh builder which hasn't reached a stable release yet?

Besides that, it wouldn't solve the issue with import type ;-)

Reply:
Welcome to open source, this isn't enterprise we can use tools that aren't perfect.

> Besides that, it wouldn't solve the issue with import type ;-)
And yet it does... I literally use this at work. Please stop trying to state facts on software you don't use.